src/java/org/apache/cassandra/db/DeletionTime.java

```diff
@@ -221,7 +222,11 @@ public DeletionTime deserialize(DataInputPlus in) throws IOException
         {
             int flags = in.readByte();
             if ((flags & IS_LIVE_DELETION) != 0)
+            {
+                if ((flags & 0xFF) != IS_LIVE_DELETION)
+                    throw new CorruptSSTableException(new Exception("Corrupted sstable. Invalid flags found deserializing DeletionTime"), "");
```

Comment:
Code as low level as this should not throw `CorruptSSTableException`. Make it an `AssertionError` or `IOException`, so that the calling code can catch it, add the sstable name to the exception, and, most importantly, mark it as corrupt.

Reply:
Switched to IOException